kbase-extension/static/kbase/js/widgets/function_output/modeling/kbaseTabTable.js

```diff
@@ -185,23 +182,23 @@ define([
             }
 
             // takes table spec and prepared data, returns datatables settings object
-            this.getTableSettings = function (tab, data) {
+            this.getTableSettings = function (tab) {
                 const tableColumns = getColSettings(tab);
 
                 const settings = {
                     dom: '<"top"lf>rt<"bottom"ip><"clear">',
                     aaData: self.obj[tab.key],
                     aoColumns: tableColumns,
-                    language: { search: '_INPUT_', searchPlaceholder: 'Search ' + tab.name },
+                    language: {search: '_INPUT_', searchPlaceholder: 'Search ' + tab.name},
+                };
+
+                const fnDrawCallback = () => {
+                    newTabEvents(tab.name);
                 };
 
                 // add any events
                 for (let i = 0; i < tab.columns.length; i++) {
-                    const col = tab.columns[i];
-
-                    settings.fnDrawCallback = function () {
-                        newTabEvents(tab.name);
-                    };
+                    settings.fnDrawCallback = fnDrawCallback;
```

Comment:
what's the iterator being used for here?

Reply:
That looks like maybe an old bug / old oversight. Part of what needs to be refactored here in this whole package. I think that's out of scope for now.